apache/trafficcontrol
# Add service category feature to TO and TP

## What does this PR (Pull Request) do?

- [x] This PR is not associated with any existing issue.

This PR enables the user to associate a delivery service with a specific service category. The user can create, add, update, delete categories as necessary and the value will be used in combination with the xmlid to set the HTTP header "X-CDN-SVC" 


## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control Client Go
- Traffic Ops
- Traffic Ops ORT - atstccfg
- Traffic Portal

## What is the best way to verify this PR?
1. Validate database changes 
       - New table service category with columns (id, name, last_updated)
       - service_category column added to delivery service table
2. Verify in TP that you can:
       - create a new delivery service with a service category
       - Add, update, delete, get service categories
3. Run the ORT script and verify that the header is being set properly 
        X-CDN-SVC "{xmlid}|{service-category}"

## The following criteria are ALL met by this PR



## Additional Information

<img width="222" alt="Screen Shot 2020-03-18 at 9 41 41 AM" src="https://user-images.githubusercontent.com/61031690/76981730-9a6d6b00-6900-11ea-97e5-87d24227e7f2.png">

<img width="1429" alt="Screen Shot 2020-03-18 at 1 19 57 PM" src="https://user-images.githubusercontent.com/61031690/76998967-8a16b980-691b-11ea-950a-5818d1e80b7a.png">

<img width="1674" alt="Screen Shot 2020-03-20 at 3 09 57 PM" src="https://user-images.githubusercontent.com/61031690/77207718-b8cd9500-6abf-11ea-9da7-784bb87930e7.png">

<img width="1678" alt="Screen Shot 2020-03-20 at 3 10 07 PM" src="https://user-images.githubusercontent.com/61031690/77207726-bcf9b280-6abf-11ea-9f36-22c4077fd69c.png">


<img width="1653" alt="Screen Shot 2020-03-18 at 1 21 07 PM" src="https://user-images.githubusercontent.com/61031690/76998989-96027b80-691b-11ea-8fc4-d07689589c8c.png">
